lib/sponsor_users.rb

```diff
@@ -0,0 +1,121 @@
+require_relative 'phone_number'
+require_relative 'email_address'
+
+class SponsorUsers
+  def initialize(user_model:)
+    @user_model = user_model
+  end
+
+  def execute(sponsees, sponsor)
+    sponsor_address = Mail::Address.new(sponsor).address
+
+    return unless EmailAddress.authorised_email_domain?(sponsor_address)
+
+    validate_sponsees(sponsees)
+    invite_sponsees(sponsor, sponsor_address)
+    send_confirmation_email(sponsor_address)
+  end
+
+private
+
+  def invite_sponsees(sponsor, sponsor_address)
+    valid_sponsees.each do |sponsee|
+      if is_valid_email(sponsee)
+        sponsee_address = Mail::Address.new(sponsee).address
+        sponsor_email(sponsor, sponsor_address, sponsee_address)
+      else
+        sponsee_phone_number = PhoneNumber.internationalise_phone_number(sponsee)
+        sponsor_phone_number(sponsor_address, sponsee_phone_number)
+      end
+    end
+  end
+
+  attr_reader :user_model, :valid_sponsees
+
+  def validate_sponsees(sponsees)
+    @valid_sponsees = sponsees.select do |sponsee|
```

Comment:
As this has an `attr_reader` I don't think you need the @ prefix. Also I think when it's accessed earlier it doesn't so should probably pick one style